### PR TITLE
Support auto transactions in SaveChanges(Async)

### PR DESCRIPTION
**Pull request by damieng**
Implements EF-81.

This is for automatic transactions only. Explicit transactions require isolation levels and save-points which are on the backlog for consideration in a future release.

### CODE REVIEW

File: src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs [hunk 1]
```diff
@@ -64,10 +89,143 @@ public override int SaveChanges(IList<IUpdateEntry> entries)
...
+
+        await transaction.CommitAsync(cancellationToken).ConfigureAwait(false);
+        return result;
     }
 
+    private MongoTransaction StartTransaction(IClientSession session, bool async)
+    {
+        var context = _currentDbContext.Context;
+        var startTime = DateTimeOffset.UtcNow;
+        var stopwatch = new Stopwatch();
```
Comment (adelinowona) [inline src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs:145]: Check if creating stopwatch starts it?

File: src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs [hunk 2]
```diff
@@ -64,10 +89,143 @@ public override int SaveChanges(IList<IUpdateEntry> entries)
...
+        {
+            await transaction.RollbackAsync(cancellationToken).ConfigureAwait(false);
+            throw;
+        }
+
+        await transaction.CommitAsync(cancellationToken).ConfigureAwait(false);
+        return result;
     }
 
+    private MongoTransaction StartTransaction(IClientSession session, bool async)
```
Comment (adelinowona) [inline src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs:141]: Consider making this a factory method on mongo transaction

**Review by adelinowona — APPROVED** [review]
LGTM. Added some comments that will be addressed in another PR.

File: src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs (hunk 2 above)
Comment (damieng) [reply src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs]: Done.

File: src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs (hunk 1 above)
Comment (damieng) [reply src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs]: Done.

**Pull request merged**